src/modules/Dashboard/features/Participant/Assignments/AssignmentsTab/AssignmentsTab.hooks.tsx

```diff
@@ -386,8 +386,14 @@ export const useAssignmentsTab = ({
         onSubmitHandler={() => navigateToData(selectedActivityOrFlow, selectedTargetSubjectId)}
       />
 
-      {showExportData && (
-        <DataExportPopup
+      {appletData && (
+        <ExportDataSetting
+          isExportSettingsOpen={showExportData}
+          onExportSettingsClose={() => setShowExportData(false)}
+          onDataExportPopupClose={() => {
+            setSelectedActivityOrFlow(undefined);
+            setSelectedTargetSubjectId(undefined);
+          }}
```

Comment:
I'm curious why in your PR notes you make an exception that on the PDP, we shouldn't expect the "Include activity flow history" checkbox to be hidden for activities. Can't we still do that here with the below change? Maybe I'm missing something.

```suggestion
          }}
          supportedSupplementaryFiles={
            selectedActivityOrFlow?.isFlow ? undefined : ['scheduleHistory']
          }
```

Reply:
Similarly, there doesn't seem to be any distinction between whether the currently viewed data is for an activity or for a flow in DataViz – `RespondentDataHeader.tsx`:

https://github.com/ChildMindInstitute/mindlogger-admin/blob/f4e44f36a064dea4725e2ebb846625a4f5d31635/src/modules/Dashboard/features/RespondentData/RespondentDataHeader/RespondentDataHeader.tsx#L184-L187

Could we not add this line there?
```tsx
              supportedSupplementaryFiles={activityFlowId ? undefined : ['scheduleHistory']}
```

Reply:
Yes, this is correct. Based on how I defined the type, the values in this array can't be dynamic. For some reason I then ended up conflating this with dynamically passing the array, but that's totally fine because the values are still static.

Thanks for pointing this out, this was a pretty silly miss 😅

About the missing filters on Dataviz, I guess we could add those in. I think that may fix [M2-9089](https://mindlogger.atlassian.net/browse/M2-9089) as well